src/snowflake/snowpark/mock/_plan.py

```diff
@@ -404,12 +405,31 @@ def handle_udf_expression(
     expr_to_alias: Optional[Dict[str, str]],
     current_row=None,
 ):
-    udf = analyzer.session.udf._registry.get(exp.udf_name)
+    udf_registry = analyzer.session.udf._registry
+    udf_name = exp.udf_name
 
-    if udf is None:
-        raise SnowparkSQLException(
-            f"[Local Testing] udf {exp.udf_name} does not exist."
-        )
+    if udf_name not in udf_registry:
+        raise SnowparkSQLException(f"[Local Testing] udf {udf_name} does not exist.")
+
+    frozen_sys_module_keys = set(sys.modules.keys())
```

Comment:
I think this should be handled with a try/finally maybe in a context manager. I think there are scenarios where this code could throw a recoverable error and we should be sure the path isn't messed up in that case.

Reply:
+1